### ISSUE TITLE
Edge may think it's unable to update when 0.7.3.0 is installed.

### ISSUE THREAD

**Issue by RealityRipple** [rcmaehl/MSEdgeRedirect#239]
### Preflight Checklist

- [X] Microsoft Edge is still installed (see issue 26)
- [X] Running microsoft-edge:https://google.com redirects successfully
- [X] Microsoft Edge is selected in any "How do you want to open this?" box (if applicable)

### Install Type

New Install

### Install Mode

Active Mode

### Steps to reproduce

This problem seems to occur on both New Installs and Updates, and requires a repair of Edge after uninstalling MSEdgeRedirect to fix.

- Install MSEdgeRedirect
- Open the Edge Menu
- Go to Help and Feedback
- Click About Microsoft Edge

### ✔️ Expected Behavior

Edge should check for updates.

### ❌ Actual Behavior

An error message occurs: "An error occurred while checking for updates: We couldn't create that component (error code 1: [0x80004005](https://go.microsoft.com/fwlink/?linkid=2068169&0x80004005) -- system level)."

### Microsoft Windows version

22H2 Build 19045.2486

**Comment by rcmaehl** [comment]
Yeah, I see that. Did it actually prevent any updates on your end? I installed Edge 108 with MSEdgeRedirect installed and it still prompted for an update despite the error.

![image](https://user-images.githubusercontent.com/716581/212748175-e82df02c-68e0-4dcf-93c8-c2756b693862.png)

**Comment by rcmaehl** [comment]
Regardless, I'll look into mitigating this error message from popping up.

**Comment by RealityRipple** [comment]
I will check that when the next update is available and report back.

**Comment by Remzi1993** [comment]
I got the same error! Therefore also occurring on an update. This is a major show stopper and needs to be given priority over anything else!

**Comment by rcmaehl** [comment]
I've confirmed this is occurring with @MarcGuiselin's wedge as well.

**Comment by rcmaehl** [comment]
I've got it manually running the updater in the meantime.

**Comment by MarcGuiselin** [comment]
> I've got it manually running the updater in the meantime.

That's probably not necessary.

So far this error message hasn't been an issue for me personally. Edge has no issues updating. I've been keeping tabs on my Edge install version ever since getting the latest release of [Wedge](https://github.com/MarcGuiselin/wedge) working and it has updated twice. Wrote a logger for flags being sent to the Edge executable, errors, exit codes, etc but there doesn't seem to be any issues. Nothing being logged in system events either.

Error seems to be caused only by executing Edge through the junction point. If you remove ifeo and execute Edge directly from its file path, the error disappears... 

I did some digging into chromium source, tracing back from the error message. Whenever you open the help page, the [`VersionUpdaterWin`](https://github.com/chromium/chromium/blob/main/chrome/browser/ui/webui/help/version_updater_win.cc) component runs a check to see if it can install an update (regardless of if one actually exists) to show relevant prompts. Among other things, it checks that the browser's execution path matches expected user or machine install path values. If they don't match this cascades to an E_FAIL (0x80004005) error [here](https://github.com/chromium/chromium/blob/main/chrome/browser/google/google_update_win.cc#L541-L544).

Since we aren't directly launching Edge from its install directory, the help page thinks Edge is installed in an invalid location and shows that error. It doesn't cause any issues anywhere else I think.

Edge might possibly never prompt for an update if it's left open a while. Testing this right now in a vm. But Edge periodically receives updates automatically via Windows anyway, so I don't think this is an issue.

**Comment by RealityRipple** [comment]
New Edge Update today according to the release notes site; still no option in the menu. I'll keep an eye on the version number and see if it updates in the background at some point.

**Comment by MarcGuiselin** [comment]
I can confirm edge has no issues downloading a new version (without closing edge or restarting my vm) and will prompt for an install once one is available. As soon as I closed edge it updated immediately.

![2023-02-04 17_32_36 - Remote Desktop Connection](https://user-images.githubusercontent.com/24906387/216778848-8ed5354a-4ea7-4a08-a431-bcf91c3eaa98.png)

This confirms what I found digging through the code. The directory check is only really ever made by the UI and pretty much serves no purpose. I suggest this issue be set to low priority.

**Comment by rcmaehl** [comment]
> I can confirm edge has no issues downloading a new version (without closing edge or restarting my vm) and will prompt for an install once one is available. As soon as I closed edge it updated immediately.
> 
> This confirms what I found digging through the code. The directory check is only really ever made by the UI and pretty much serves no purpose. I suggest this issue be set to low priority.

Maybe I should make a pull request on Chromium to fix this for IFEO and Chrome Portable users. Although I don't think they'll be entirely convinced.

**Comment by RealityRipple** [comment]
Update just showed up here as well. Just took an extra day or two.

**Comment by MarcGuiselin** [comment]
> Maybe I should make a pull request on Chromium to fix this for IFEO and Chrome Portable users. Although I don't think they'll be entirely convinced.

I was thinking that too. Even if that check is only there for legacy reasons, without a reproducible bug (outside our use-case) I don't think they'll be inclined to fix it.

I did some more digging and I found that a symlink works just as fine as a junction point. Except now, we can symlink directly to msedge rather than its directory. This method also does away with the install error if the symlink used to launch edge is placed in the same install directory. So we should look into that for a solution.

Basically this would bring us back to what MSEdgeRedirect was doing before, except using a symlink in place of msedge_no_ifeo.

**Comment by rcmaehl** [comment]
How were you executing the symlink? ShellExecute? I was just getting an infinite loop.

**Comment by MarcGuiselin** [comment]
Via rust's `Command::new`, which is just `CreateProcess` under the hood.

**Comment by purevertigo** [comment]
Is this going to be fixed soon?

**Comment by rcmaehl** [comment]
> Is this going to be fixed soon?

Yep

**Comment by woodspire** [comment]
What is your definition of "soon" ? In a lifetime, 3 years can be considered "soon". But in the current context, I thought, since there doesn't seem to be any further development needed, that "soon" would mean: in the coming month (singular) ...

**Comment by MarcGuiselin** [comment]
> What is your definition of "soon" ?

"Soon" is whenever he can get to it. Unless you'd want to contribute to the software to fix this issue yourself in the coming month (instead of being unconstructive). This work is done pro bono. No one owes you anything. As I've discovered, and what has been thoroughly discussed in this thread, is that the priority for fixing this bug is low. It's just a visual bug in edge with zero consequence for actually receiving software updates.

**Comment by rcmaehl** [comment]
> What is your definition of "soon" ? In a lifetime, 3 years can be considered "soon". But in the current context, I thought, since there doesn't seem to be any further development needed, that "soon" would mean: in the coming month (singular) ...

Should be fixed in the latest dev build. As Marc said it's a low priority visual bug from Edge being confused. My rough update outline from the latest release is as follows:

0.8.0.0 (Spring):

Per User Active Mode Option
Installer and Updater Logic Overhaul
Service Mode Quality of Life Improvements

0.9.0.0 (Summer):

UI Overhaul
Code Freeze
Wiki and other documentation updated

1.0.0.0 (End of Summer):

Code Refactor/Rewrite
PortableApps.com Version Release

1.0.1.0 (Probably the next day due to bugs):

Bug fixes
